provider/cmd/pulumi-gen-eks/main.go

```diff
@@ -300,12 +306,18 @@ func generateSchema() schema.PackageSpec {
 						Description: "Optional mappings from AWS IAM users to Kubernetes users and groups.",
 					},
 					"vpcCniOptions": {
-						TypeSpec: schema.TypeSpec{Ref: "#/types/eks:index:VpcCniOptions"},
+						TypeSpec: schema.TypeSpec{
+							Ref:   "#/types/eks:index:VpcCniOptions",
+							Plain: true,
+						},
 						Description: "The configuration of the Amazon VPC CNI plugin for this instance. Defaults are " +
 							"described in the documentation for the VpcCniOptions type.",
 					},
 					"useDefaultVpcCni": {
-						TypeSpec:    schema.TypeSpec{Type: "boolean"},
+						TypeSpec: schema.TypeSpec{
+							Type:  "boolean",
+							Plain: true,
+						},
```

Comment:
This is plain in the implementation:

https://github.com/pulumi/pulumi-eks/blob/c6f1fce98fa2683abe7ffbad7f166da99928fb83/nodejs/eks/cluster.ts#L987-L990